flake.nix

```
@@ -0,0 +1,145 @@
+{
+  description =
+    "Quarkdown CLI - A modern Markdown-based typesetting system with superpowers";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+
+        # Quarkdown version - using latest stable release
+        version = "1.4.0";
+
+        # Build Quarkdown from source
+        quarkdown = pkgs.stdenv.mkDerivation rec {
+          pname = "quarkdown";
+          inherit version;
+
+          src = pkgs.fetchFromGitHub {
+            owner = "iamgio";
+            repo = "quarkdown";
+            rev = "v${version}";
+            sha256 = "sha256-9WQ3+VjlZmlDcHwSVOWD/ZGJfRCnXdKF0humGq2d8CU=";
+          };
+
+          nativeBuildInputs = with pkgs; [ gradle_8 jdk17 makeWrapper unzip ];
+
+          buildInputs = with pkgs; [ jdk17 ];
+
+          # Gradle needs to download dependencies
+          GRADLE_USER_HOME = ".gradle";
+
+          configurePhase = ''
+            runHook preConfigure
+
+            # Set up Gradle home
+            export GRADLE_USER_HOME=$(mktemp -d)
+
+            # Ensure we're using JDK 17 to avoid compatibility issues
+            export JAVA_HOME=${pkgs.jdk17}
+            export PATH=${pkgs.jdk17}/bin:$PATH
+
+            runHook postConfigure
+          '';
+
+          buildPhase = ''
+            runHook preBuild
+
+            echo "Building Quarkdown distribution with Gradle..."
+            ./gradlew --no-daemon --console=plain -Dorg.gradle.java.home=${pkgs.jdk17} quarkdown-cli:distZip
```

```suggestion
            gradle --no-daemon --console=plain -Dorg.gradle.java.home=${pkgs.jdk17} quarkdown-cli:distZip
```

This needs to use the Nix supplied `gradle`, not the wrapper script that fetches it's own.

---

Already had made the change! :)